components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/extension/AbstractWorkflowRequestHandler.java

```diff
@@ -209,4 +214,23 @@ private boolean isWorkflowCompleted() {
             return true;
         } else return false;
     }
+
+    /**
+     * We can check whether the current event type is already associated with
+     * at-least one association or not by using isAssociated method.
+     *
+     * @return Boolean value for result of isAssociated
+     * @throws WorkflowException
+     */
+    public boolean isAssociated() throws WorkflowException{
```

Comment:
Seems like this checks if event ADD_USER associated with a workflow or not, but since we can associate with a given Boolean expression this may not correctly return if currently operating event is associated with a workflow or not, rather it'll check if there at least 1 association for the event "ADD_USER",
for example let's assume we have engaged workflows for adding users who are assigned to 'admin' role only, but for a adding user who don't have 'admin' role, this function will still return isAssociate=true, since there is a ADD_USER engagement for different type of ADD_USER, and still be added through call back even though there are no workflows associated.
So this fix may reduce the damage, but will not fix the issue completely.